File: agent/eterion_agent/models/capabilities.py
"""Explicit model capability metadata.

OpenAI-compatible endpoints share a request shape, but reasoning and tool-call
streams still vary by provider. Unknown capabilities stay disabled until they
are covered by a provider contract test.
"""

from __future__ import annotations

from dataclasses import dataclass
from enum import StrEnum


class CapabilityStatus(StrEnum):
    VERIFIED = "verified"
    UNKNOWN = "unknown"
    UNSUPPORTED = "unsupported"


@dataclass(frozen=True, slots=True)
class ModelCapabilities:
    text_streaming: CapabilityStatus = CapabilityStatus.VERIFIED
    thinking_streaming: CapabilityStatus = CapabilityStatus.UNKNOWN
    tool_calling: CapabilityStatus = CapabilityStatus.UNKNOWN
    parallel_tool_calls: CapabilityStatus = CapabilityStatus.UNKNOWN

File: agent/src/models/capabilities.ts
export type CapabilityStatus = 'verified' | 'unknown' | 'unsupported';

export interface ModelCapabilities {
  textStreaming: CapabilityStatus;
  thinkingStreaming: CapabilityStatus;
  toolCalling: CapabilityStatus;
  parallelToolCalls: CapabilityStatus;
}

/** OpenAI-compatible 不代表 Tool/Thinking 行为兼容，未知能力不能默认开启。 */
export const DEFAULT_MODEL_CAPABILITIES: Readonly<ModelCapabilities> = {
  textStreaming: 'verified',
  thinkingStreaming: 'unknown',
  toolCalling: 'unknown',
  parallelToolCalls: 'unknown',
};
